### PR TITLE
feat(color-picker): preset colors support collapsing at initialization

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     feat(color-picker): preset colors support collapsing at initialization      |
| 🇨🇳 Chinese |      color-picker 预设颜色支持默认折叠模式     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at f88d3a8</samp>

This pull request adds a new feature to the color picker component, which allows the user to specify the initial folding state of the presets using the `collapsed` property. It also refactors the code for generating and managing the collapse panel keys, and adds tests and documentation for the new feature.

### <samp>🤖 Generated by Copilot at f88d3a8</samp>

*  Add `collapsed` property to `presets` prop of color picker component, allowing users to control the initial folding state of the presets ([link](https://github.com/ant-design/ant-design/pull/45607/files?diff=unified&w=0#diff-819859f3d30fc8ec12a8ef1b55ac1d75b4b1d600eb1cee989efd5af57255b560L56-R56), [link](https://github.com/ant-design/ant-design/pull/45607/files?diff=unified&w=0#diff-9d1ff106a032e484c8f982279c4e1295d23f69904a26e020e99310c446e93249L57-R57), [link](https://github.com/ant-design/ant-design/pull/45607/files?diff=unified&w=0#diff-83dba3026e7dc41ad64d3a2f4d484eeb085f0c3798dc2316fbb90d23520fca6eR14-R18))
*  Generate unique keys for collapse panels based on preset labels using `genCollapsePanelKey` helper function ([link](https://github.com/ant-design/ant-design/pull/45607/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822R37-R38), [link](https://github.com/ant-design/ant-design/pull/45607/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822L56-R64))
*  Filter out collapsed presets from default active keys using `genCollapsePanelKey` function ([link](https://github.com/ant-design/ant-design/pull/45607/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822L47-R55))
*  Test color picker behavior with collapsed and non-collapsed presets in `components/color-picker/__tests__/index.test.tsx` ([link](https://github.com/ant-design/ant-design/pull/45607/files?diff=unified&w=0#diff-aa06460351e8630f305cf6509e8137984e0f167ab2560da22874033189234fb4R228-R260))

### CODE REVIEW

File: components/color-picker/components/ColorPresets.tsx [hunk 1]
```diff
@@ -44,7 +46,13 @@ const ColorPresets: FC<ColorPresetsProps> = ({ prefixCls, presets, value: color,
   const colorPresetsPrefixCls = `${prefixCls}-presets`;
 
   const activeKeys = useMemo<string[]>(
```
Comment (Wxh16144) [inline components/color-picker/components/ColorPresets.tsx:48]: image

```suggestion
  const activeKeys = useMemo(
```

Comment (Wxh16144) [reply components/color-picker/components/ColorPresets.tsx:48]: 还是不加防御了吧，这里就交给开发者自己双非 (!!) 处理

File: components/color-picker/interface.ts [hunk 2]
```diff
@@ -11,6 +11,11 @@ export enum ColorFormat {
 export interface PresetsItem {
   label: ReactNode;
   colors: (string | Color)[];
+  /**
+   * Whether the initial state is collapsed
+   * @since 5.11.0
+   */
+  collapsed?: boolean;
```
Comment (afc163) [inline components/color-picker/interface.ts:18]: 非受控属性，应该是 defaultCollapsed

File: components/color-picker/components/ColorPresets.tsx [hunk 3]
```diff
@@ -44,7 +46,13 @@ const ColorPresets: FC<ColorPresetsProps> = ({ prefixCls, presets, value: color,
...
-    () => presetsValue.map((preset) => `panel-${preset.label}`),
+    () =>
+      presetsValue.reduce<string[]>((acc, preset) => {
+        if (preset.collapsed !== true) {
```
Comment (RedJue) [inline components/color-picker/components/ColorPresets.tsx:51]:
```suggestion
        if (!preset.collapsed) {
```

File: components/color-picker/interface.ts (hunk 2 above)
Comment (Wxh16144) [reply components/color-picker/interface.ts:18]: 嗯嗯，确实是非受控的~

File: components/color-picker/index.zh-CN.md [hunk 4]
```diff
@@ -54,7 +54,7 @@ group:
 | destroyTooltipOnHide | 关闭后是否销毁弹窗 | `boolean` | false | 5.7.0 |
 | format | 颜色格式 | `rgb` \| `hex` \| `hsb` | `hex` | |
 | open | 是否显示弹出窗口 | boolean | - | |
-| presets | 预设的颜色 | `{ label: ReactNode, colors: Array<string \| Color> }[]` | - | |
+| presets | 预设的颜色，`5.11.0+` 支持 `defaultCollapsed` 以实现初始化时默认折叠预设 | `{ label: ReactNode, colors: Array<string \| Color>, defaultCollapsed?: boolean }[]` | - | |
```
Comment (MadCcc) [inline components/color-picker/index.zh-CN.md:57]:
```suggestion
| presets | 预设的颜色| `{ label: ReactNode, colors: Array<string \| Color>, defaultCollapsed?: boolean }[]` | - | `defaultCollapsed: 5.11.0` |
```

File: components/color-picker/index.en-US.md [hunk 5]
```diff
@@ -53,7 +53,7 @@ Common props ref：[Common props](/docs/react/common-props)
 | destroyTooltipOnHide | Whether destroy popover when hidden | `boolean` | false | 5.7.0 |
 | format | Format of color | `rgb` \| `hex` \| `hsb` | `hex` | |
 | open | Whether to show popup | boolean | - | |
-| presets | 	Preset colors | `{ label: ReactNode, colors: Array<string \| Color> }[]` | - | |
+| presets | Preset colors, `5.11.0+` Support for `defaultCollapsed` to achieve default folding during initialization | `{ label: ReactNode, colors: Array<string \| Color>, defaultCollapsed?: boolean }[]` | - | |
```
Comment (MadCcc) [inline components/color-picker/index.en-US.md:56]:
```suggestion
| presets | Preset colors | `{ label: ReactNode, colors: Array<string \| Color>, defaultCollapsed?: boolean }[]` | - | `defaultCollapsed: 5.11.0` |
```

**Review by MadCcc — APPROVED** [review]

**Pull request merged**